src/StrawberryfieldUtilityService.php

```diff
@@ -346,54 +367,48 @@ public function getCountByProcessorInSolr(EntityInterface $entity, string $proce
   /**
    * {@inheritdoc}
    */
-  public function internalFilePathIsValid(string $scheme, string $path): bool {
+  public function filePathIsValid(string $scheme, string $path): bool {
     // Empty path is always valid.
-    if(empty($path)) {
+    if (empty($path)) {
       return TRUE;
     }
-    // First check to see if the path is ok.
-    $path_valid = preg_match(self::PATHPATTERN, $path, $matches);
-    if(!$path_valid) {
-      return FALSE;
-    }
-
-    // TODO is there an easier way to determine if the file path is valid and writable?
-    //   \Drupal::service('path.validator')->isValid($path) always returns FALSE,
-    //   and \Drupal::service('stream_wrapper_manager')->isValidUri($path) always returns TRUE!
-
-    $test_path = $scheme . "://" . $path;
 
-    // FileSystemInterface::prepareDirectory with the MODIFY_PERMISSIONS flag checks to see if it's present and able to be written to.
-    $valid = \Drupal::service('file_system')->prepareDirectory($test_path, FileSystemInterface::MODIFY_PERMISSIONS);
-    if(!$valid) {
-      // If not present or able to be written to, let's check to see if we can create it.
-      $valid = \Drupal::service('file_system')->prepareDirectory($test_path, FileSystemInterface::CREATE_DIRECTORY);
-      if($valid) {
-        // Successfully created the directory, so we delete it now.
-        \Drupal::service('file_system')->rmdir($test_path);
-      }
-    }
-    return !empty($valid);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function s3FilePathIsValid(string $path): bool {
-    // Empty path is always valid.
-    if(empty($path)) {
+    $known_schemes = array_merge(['s3'], array_keys(\Drupal::service('stream_wrapper_manager')->getWrappers(StreamWrapperInterface::LOCAL)));
```

Comment:
Again, we have the OCFL toolbox I wrote to fetch all known schemes that are active and can be used to writing. Will refactor

Reply:
Ok, thanks for your patience with a D8+ noob!